docs/training/apis-howto/deepspeed/autotuning.rst

```diff
@@ -0,0 +1,247 @@
+two helper functions into your code and then replacing the usual experiment launching command 
+
+.. code::
+
+  det experiment create deepspeed.yaml .
+
+with
+
+.. code::
+
+  python3 -m determined.pytorch.dsat asha deepspeed.yaml .
+
+where ``deepspeed.yaml`` is the Determined configuration for a ``single`` experiment. There is no
+need to write a special configuration file to use ``dsat``.
+
+The above uses the ASHA algorithm (TODO: link to ASHA page) to tune the DS parameters and is one
+of the three search methods we currently provide:
+
+* ``asha``: adaptively searches over randomly selected DeepSpeed configurations, yielding more
+   compute resources to well-performing classes of configurations.
+
+* ``binary``: simple binary searches over the batch size for randomly-generated DS configurations.
+
+* ``random``: a search over random DeepSpeed configurations with an aggressive early-stopping
+  criteria based on domain-knowledge of DeepSpeed and the search history.
+
+DeepSpeed Autotune is built on top of Custom Searcher (TODO: link) 
+which starts up two separate experiments:
+
+* A ``single`` search runner experiment whose role is to coordinate and schedule the actual trials
+  which run model code. 
+* A ``custom`` experiment which contains the trials referenced above whose results are reported back
+  to the search runner.
+
+The logs of the ``single`` search runner experiment will contain information regarding the size of the model,
+the GPU memory available, the activation memory required per example, and an approximate computation
+of the maximum batch size per zero stage (which is used to guide the starting point for all
+autotuning searches). When a best-performing DS configuration is found, the
+corresponding ``json`` configuration file will be written to the search runner's checkpoint directory,
+along with a file detailing the configuration's corresponding metrics.
+
+The ``custom`` experiment instead holds the visualization and summary tables which outline the results for
+every Trial. Initially, a one-step profiling Trial is created to gather the above information regarding
+the model and available hardware. Subsequently, multiple short Trials are submitted which each report
+back metrics such as ``FLOPS_per_gpu``, ``throughput`` (samples/second), and latency timing information.
+
+In the following sections, we describe the specific user-code changes which must be made if using
+``dsat`` with :class:`~determined.pytorch.deepspeed.DeepSpeedTrial`, Core API, and HuggingFace
+Trainer, respectively.
+
+``DeepSpeedTrial``
+==================
+
+Determined's DeepSpeed Autotune works by writing DS configuration options to the ``overwrite_deepspeed_args``
+field of the ``hyperparameters`` dictionary that is seen by each trial. Assuming your default DS
+configuration is specified in a ``json`` file, then you only need to incorporate
+these overwrite values into your original configuration in order to take advantage of ``dsat``.
```

Comment:
Could we point to the deepspeed documentation to explain more clearly what we mean with the json files, and a quick idea of the important parameters in it are? Maybe this could be a section on its own just to make sure users are on the same page? This paragraph can be pretty confusing to someone reading this out of context.

Reply:
maybe a note with a link to deepspeed documentation where these parameters are documented? that way we don't have to maintain the content when it gets updated at the source.